gamification-evm-services/src/main/java/io/meeds/gamification/evm/service/EvmTriggerService.java

```diff
@@ -0,0 +1,140 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package io.meeds.gamification.evm.service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import jakarta.annotation.PostConstruct;
+
+import org.exoplatform.wallet.service.WalletAccountService;
+import org.exoplatform.wallet.model.Wallet;
+import io.meeds.gamification.evm.model.EvmTrigger;
+import io.meeds.gamification.model.EventDTO;
+import io.meeds.gamification.service.ConnectorService;
+import io.meeds.gamification.service.EventService;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
+import org.exoplatform.commons.api.persistence.ExoTransactional;
+import org.exoplatform.services.listener.ListenerService;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import static io.meeds.gamification.evm.utils.Utils.*;
+
+@Component
+public class EvmTriggerService {
+
+  private static final Log   LOG                        = ExoLogger.getLogger(EvmTriggerService.class);
+
+  public static final String GAMIFICATION_GENERIC_EVENT = "exo.gamification.generic.action";
+
+  @Autowired
+  private ConnectorService connectorService;
+
+  @Autowired
+  private IdentityManager identityManager;
+
+  @Autowired
+  private ListenerService listenerService;
+
+  @Autowired
+  private EventService eventService;
+
+  @Autowired
+  private WalletAccountService walletAccountService;
+
+  private ExecutorService executorService;
+
+  public EvmTriggerService(ConnectorService connectorService,
+                               IdentityManager identityManager,
+                               ListenerService listenerService,
+                               EventService eventService) {
+    this.connectorService = connectorService;
+    this.identityManager = identityManager;
+    this.listenerService = listenerService;
+    this.eventService = eventService;
+  }
+
+  @PostConstruct
+  public void initialize() {
+    QueuedThreadPool threadFactory = new QueuedThreadPool(5, 1, 1);
+    threadFactory.setName("Gamification - Evm connector");
+    executorService = Executors.newCachedThreadPool(threadFactory);
```

Comment:
We should `shutdown` the executor. Thus please add a `@PreDestroy` annotated method to stop it when the server is stopped.

```suggestion
    executorService = Executors.newCachedThreadPool(threadFactory);
```